### DISCUSSION TITLE
Renderer process crashes with virtualenv install on Gentoo

### DISCUSSION THREAD

**Discussion by RadoslavL** [qutebrowser/qutebrowser#6978] [Q&A] [answered]
**Version info**:

I am unable to open qute://version in the browser
**Does the bug happen if you start with `--temp-basedir`?**:
Yes
**Description**
I get a white screen, that's flickering between gray and white and The errors in the logs say "Renderer process crashed (status 31)"
**How to reproduce**
 It is not caused by any specific website. It happens on all of the websites I tried. Even on my local nginx php server.

**Comment by The-Compiler** [comment]
This is likely an issue with your system (and/or QtWebEngine/Chromium).
Could you run with `--version` to get the version info, if needed with `--debug-flag avoid-chromium-init`? What OS/distribution is this on? Could you check whether you get similar issues with KDE Falkon (another browser using QtWebEngine) and/or Chromium?

**Comment by RadoslavL** [comment]
qutebrowser v2.4.0
Git commit: 81e1fb2cc on master (2022-01-25 23:27:06 +0100)
Backend: QtWebEngine 5.15.2, based on Chromium 83.0.4103.122
Qt: 5.15.2

I am using Gentoo and it does not happen with Chrome. Not sure about Chromium though. The browser was working yesterday and I am not sure I did anything, that could've caused this. I tried to install LFS yesterday and that could've caused it, but not sure about that.

**Comment by The-Compiler** [comment]
Would be good if you could try with Falkon, or at least our [minimal testbrowser](https://github.com/qutebrowser/qutebrowser/blob/master/scripts/testbrowser/testbrowser_webengine.py). I doubt this is specific to qutebrowser.

**Comment by RadoslavL** [comment]
Is qutebrowser with webkit considered stable? I installed webkit-gtk in order to make lutris work, but i do not think i would have the time and CPU power to build qtwebengine from source.

**Comment by The-Compiler** [comment]
I don't see why you would need to, given that you already are running qutebrowser with QtWebEngine?

qutebrowser uses QtWebKit (not WebKitGTK), and that's still based on a WebKit from 2016, so I wouldn't exactly call it stable (and I'd avoid using it).

**Comment by RadoslavL** [comment]
Falkon works with flatpak. Not sure if installing it with portage would make a difference, but i avoided doing that, because i do not have qtwebengine installed on my machine. I used the virtualenv installation before and i don't really want to install the distro version of Falkon. I might do it when i have more free time.

**Comment by The-Compiler** [comment]
Could you try with `.venv/bin/python scripts/testbrowser/testbrowser_webengine.py` in that case?

**Comment by RadoslavL** [comment]
It opens a blank white window with no errors. It just displays "Loading progress: 100%" and nothing changes except a quick flash on the window.

**Comment by The-Compiler** [comment, marked as answer]
Same problem then, other than the testbrowser not handling/printing the renderer process crash. Given that this is an issue on your system rather than qutebrowser, I'm going to convert this into a discussion.

One more thing you could try is running qutebrowser with `--qt-flag disable-seccomp-filter-sandbox`. This disables one of Chromium's security layers, but would be the most likely culprit if this started happening after an update of e.g. glibc. If that's the case, I'm afraid there is not much you can do other than either disabling the sandboxing, or using Gentoo's QtWebEngine (which is newer than what you get via a prebuilt Qt binary).

**Reply by RadoslavL** [reply]
@The-Compiler It works with that flag. Will, using it, cause a security thread? I didn't know this existed and had no idea there was a reason to even put on a security layer.

**Reply by The-Compiler** [reply]
Yes: It disables one of Chromium's [sandboxing layers](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/linux/sandboxing.md), notably the seccomp-BPF sandbox, which has a list of syscalls the renderer process is allowed to make. This means that, if someone exploits a security bug in the underlying Chromium, there is one layer less keeping them away from your system.

Based on a comment in the [`glibc` package](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-libs/glibc/glibc-2.34-r7.ebuild#n909), perhaps disabling the `clone3` useflag for `glibc` might help as a (probably better) workaround, until you can move to a newer QtWebEngine (either Gentoo's, or via qutebrowser supporting Qt 6, see #5395).

**Reply by RadoslavL** [reply]
@The-Compiler I disabled the clone3 use flag and it works fine now.

**Reply by RadoslavL** [reply]
I found out, what was the reason for this issue to happen! I updated glibc on my host system in order to make gcc work without booting or chrooting into the system. The installed version on the host and the installation must be the same to make it work.
